### PR TITLE
Thai/include deleted times

### PR DESCRIPTION
**Pull request by athai**
Tests for the include_deleted times parameter have been written.

test_data has been modified to include deleted time entries and associated timesactivities

### CODE REVIEW

File: tests/fixtures/test_data.json [hunk 1]
```diff
@@ -268,6 +296,16 @@
       "id": 5,
       "time": "times:3",
       "activity": "activities:1"
+    },
+    {
+      "id": 6,
+      "time": "times:4",
```
Comment (Kennric) [inline tests/fixtures/test_data.json]: Should this be associated with time 5, and the next one 6, instead of 4 and 5?

File: tests/times.js [hunk 2]
```diff
@@ -1436,6 +1436,242 @@ module.exports = function(expect, request, baseUrl) {
...
+        id: 5,
+        uuid: 'b6ac75fb-7872-403f-ab71-e5542fae4212',
+        revision: 1,
+        deleted_at: '2015-05-02',
+      },
+      {
+        duration: 12,
+        user: 'patcht',
+        project: ['pgd'],
+        activities: ['meeting'],
```
Comment (Kennric) [inline tests/times.js]: Is this activity correctly associated in the test data?

File: tests/fixtures/test_data.json (hunk 1 above)
Comment (athai) [inline tests/fixtures/test_data.json]: I don't think so. Why would it be associated with time 5 instead?

**Comment by Kennric** [comment]
I think these look pretty good, I'd get a second opinion and then start implementing. +1

**Comment by ElijahCaine** [comment]
After a (quick) review these tests look good and pretty consistent with our other tests. +1

File: tests/fixtures/test_data.json (hunk 1 above)
Comment (LyonesGamer) [inline tests/fixtures/test_data.json]: Specifically, the times table is 1-indexed like all SQL databases, but the fixtures syntax used here is 0-indexed and based on the order it's entered into the DB (i.e. the order listed in the file). It's weird.

**Comment by Kennric** [comment]
This looks pretty good. +1

**Comment by athai** [comment]
You can currently query for deleted times that are specific to a project, activity, or user. Should I also add tests that check whether results are narrowed further by combining parameters? That is, tests that check whether requests such as:
* ``GET /times?user=:user&activity=:activityslug&project=:projectslug&include_deleted`` 
* ``GET /times?user=:user&activity=:activityslug&include_deleted``
*  ``GET /times?user=:user&project=:project&include_deleted`` 

returns the correct result. If so, how many combined parameters should I check for?

File: src/activities.js [hunk 3]
```diff
@@ -13,10 +13,11 @@ module.exports = function(app) {
     /* Only query for activities that have not been soft-deleted
      * If the include_deleted param is true, query for all activities
      * regardless of 'deleted' status */
-    if (req.query.include_deleted) {
+    if (req.query.include_deleted === 'true' ||
+        req.query.include_deleted === '') {
       activitiesQ = knex('activities');
     } else {
-      activitiesQ = knex('activities').whereNull('deleted_at');
+      activitiesQ = knex('activities').where({deleted_at: null});
```
Comment (ElijahCaine) [inline src/activities.js]: Is this different or just more consistent with how we've been doing things?

**Comment by ElijahCaine** [comment]
Tests pass, implementation looks straight forward and is easy to understand. +1 from me.

File: src/activities.js (hunk 3 above)
Comment (athai) [inline src/activities.js]: Just more consistent with how it's done elsewhere.

**Comment by Kennric** [comment]
Further param combo tests are being added.

**Comment by Kennric** [comment]
I think you have everything tested, but there is a lot of code there, please also have @LyonesGamer or @ElijahCaine look it over. +1

**Comment by ElijahCaine** [comment]
Everything looks good to me. Very thorough. :+1:

**Pull request merged**